Source/Core/Form.swift

```diff
@@ -172,7 +172,17 @@ extension Form: MutableCollection {
 
     public subscript (_ position: Int) -> Section {
         get { return kvoWrapper.sections[position] as! Section }
-        set { kvoWrapper.sections[position] = newValue }
+        set {
+            kvoWrapper.sections[position] = newValue
+            if position < kvoWrapper._allSections.count {
+                // Remove an existing section from the form
+                kvoWrapper._allSections[position].willBeRemovedFromForm()
+                kvoWrapper._allSections[position] = newValue
```

Comment:
S1, S2, S5                            <-- S6 index is 3

S1, S2, S3(H), S4(H), S5    <-- S6 index is 5 here